### ISSUE TITLE
Performance problem of ParaViewDataCollection

### ISSUE THREAD

**Issue by Liu0813** [mfem/mfem#2120]
Hi,
I am developing a one-dimensional transient solver based on MFEM, but the one-dimensional results of MFEM can't be visualized by VisIt. Thus, the class `ParaViewDataCollection` is used in my solver. However, the test results show that the output of the ParaView file will seriously affect the performance. A test code is given below,
```c++
#include "mfem.hpp"

using namespace std;
using namespace mfem;

int main(int argc, char *argv[])
{
   const char *mesh_file = "./mesh-1.mesh";
   const char *basename = "results";

   Mesh *mesh = new Mesh(mesh_file, 1, 1);
   int dim = mesh->Dimension();

   mesh->UniformRefinement();

   // 5. Define a finite element space on the mesh. Here we use continuous
   //    Lagrange finite elements of the specified order. If order < 1, we
   //    instead use an isoparametric/isogeometric space.
   FiniteElementCollection *fec;
   fec = new H1_FECollection(2, dim);
   FiniteElementSpace *fespace = new FiniteElementSpace(mesh, fec);

   GridFunction x1_gf(fespace);
   x1_gf = 1;
   GridFunction x2_gf(fespace);
   x2_gf = 2;
   GridFunction x3_gf(fespace);
   x3_gf = 1;

   VisItDataCollection paraview_dc(basename, mesh);
   paraview_dc.RegisterField("X1",  &x1_gf);
   paraview_dc.RegisterField("X2", &x2_gf);
   paraview_dc.RegisterField("X3", &x3_gf);

   StopWatch timer;
   for (int i = 0; i < 10000; i++)
   {
      timer.Start(); 
      paraview_dc.SetCycle(i);
      paraview_dc.SetTime(i * 1.0e-4);
      paraview_dc.Save();
      timer.Stop();
      cout << "step = " << i << ", time = " << timer.RealTime() << " s" << endl;
      timer.Clear();
   }

   delete fespace;
   delete fec;
   delete mesh;

   return 0;
}
```
It was shown that the time it takes for `ParaViewDataCollection` to output data once will be longer and longer as the time step increases. The time taken for each output is as follows:
![image](https://user-images.githubusercontent.com/30075822/112158359-0cb71080-8c23-11eb-82b1-820385e76e5e.png)

When the number of output files continues to increase, the time is unacceptable. Maybe that is where MFEM needs to be improved. But for this version, is there other output format for 1d data?

**Comment by pazner** [comment]
This is a bug in `GeometryRefiner` that only appears for 1D meshes. It is fixed in #2122. Is the output sufficiently fast for you with that fix?

(Your code snippet has `VisItDataCollection paraview_dc(basename, mesh);`. I assume that should be changed to `ParaViewDataCollection`)

**Comment by Liu0813** [comment]
@pazner, thank you for your reply, the fixed code is sufficiently fast.
